temporalio/worker/_workflow_instance.py

```diff
@@ -287,7 +288,17 @@ def activate(
                 # be checked in patch jobs (first index) or query jobs (last
                 # index).
                 self._run_once(check_conditions=index == 1 or index == 2)
+        except temporalio.exceptions.FailureError as err:
+            # We want failure errors during activation, like those that can
+            # happen during payload conversion, to fail the workflow not the
+            # task
+            try:
+                self._set_workflow_failure(err)
```

Comment:
Making sure - this is only happening if they explicitly raise something like `ApplicationError` - not on any old random exception, correct?

Reply:
Correct, this catch clause is for `temporalio.exceptions.FailureError` only